### PR TITLE
added google tag manager hooks for logging metrics

### PR DESCRIPTION
**Pull request by thetif**
Added Google Tag Manager hooks into front end code. Added custom variables for state, a hashed version of the open APD, and the anchor tag to be sent to GTM. Configured GTM and GA to make the requested reports.

### This pull request changes...

- changed the way our application communicates with Google Analytics
- created some custom variables to also send to Google Analytics to get a clearer picture

### This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
~- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented~
~- [ ] The change has been documented~
~- [ ] Associated OpenAPI documentation has been updated~
- [ ] Changelog is updated as appropriate

### This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -86,6 +87,7 @@
     "stickybits": "^3.7.4",
     "tinymce": "^5.2.2",
     "updeep": "^1.2.0",
+    "uuid": "^8.3.1",
```
Comment (radavis) [inline web/package.json:90]: Is it necessary to add 'uuid' as a dependency?

File: web/src/reducers/apd.js [hunk 2]
```diff
@@ -230,6 +230,20 @@ export const getKeyPersonnel = (years = []) => ({
...
+}) => {
+  if (!id) return null;
+  return id
+    .toString()
+    .split('')
+    .map(v => v.charCodeAt(0))
+    .reduce((a, v) => (a + ((a << 7) + (a << 3))) ^ v) // eslint-disable-line no-bitwise
+    .toString(16);
+};
+
```
Comment (radavis) [inline web/src/reducers/apd.js:246]: This is interesting. Why not send the APD id?

**Review by radavis — APPROVED** [review]
Looks good to me, based on the react-gtm docs

File: web/package.json (hunk 1 above)
Comment (radavis) [reply web/package.json:90]: I see that this is used in a subsequent PR. No worries.

**Review by CSwartzHMA — APPROVED** [review]

**Pull request merged**